Python/ceval.c

```diff
@@ -3173,6 +3174,25 @@ _PyEval_EvalFrameDefault(PyThreadState *tstate, PyFrameObject *f, int throwflag)
                     la = &co_opcache->u.la;
                     if (la->type == type && la->tp_version_tag == type->tp_version_tag)
                     {
+                        // Hint >= 0 is a dict index; hint < 0 is an inverted slot index
+                        if (la->hint < 0) {
+                            /* Even faster path -- slot hint */
+                            Py_ssize_t offset = ~la->hint;
+                            char *addr = (char *)owner + offset;
+                            res = *(PyObject **)addr;
+                            if (res != NULL) {
+                                Py_INCREF(res);
+                                SET_TOP(res);
+                                Py_DECREF(owner);
+                                DISPATCH();
+                            }
+                            else {
+                                Py_DECREF(owner);
+                                goto error;
```

Comment:
Perhaps we want to `OPCACHE_DEOPT_LOAD_ATTR();` here.

Reply:
But why would we? The slot is still there, it's just empty. The hint is still correct.

Reply:
Ah, right, makes sense.